### ISSUE TITLE
Used BLE sensors not connected

### ISSUE THREAD

**Issue by ccharn** [Sublimis/UrbanBiker#84]
First time user of your app. I am using Xperia XZ and HRM, Speed, Cadence sensors by Magene. Sensors easily found after scanning. Works perfectly first time. Then on subsequent uses on the next rides it takes me numerous fidling around like deletes, scans, and reassign before all of a sudden they are recognised and transmitting data. So far they never drop during entire ride once connected. Tried new batteries with the same issues. All sensors are new. Also decided to try Urban Biker on another bike with CooSpo combined speed and cadence sensor. Again first time works perfectly. Unfortunately after rebooting, the sensor is no longer connecting just like on the Magene sensors. Tried numerous deletions and re-scans before succeeding to connect again. 

Thanks you for a great app.

**Comment by Sublimis** [comment]
Hi, the most important thing to note is that sensors get connected only once tracking (recording) is active. They will not get connected nor send any data otherwise. Is it possible that this is the culprit?

If not, than please allow some time for the sensors to connect, maybe even a minute or so in the worst case. Also make sure that your phone has all the latest system updates installed.

Please let us know if that helped. Thanks for the feedback!

**Comment by ccharn** [comment]
I am well aware that tracking must be started for sensors to be connected.

UB can "find" them but once selected for use and configured, and track started, they (very often than not) cannot be connected. They can suddenly come alive 30 minutes later (or never) during a ride. But every time I succeed in getting them detected, they are rock solid stable for the whole ride.

I have tried new batteries, cleared app and Bluetooth cache, delete and re scan, etc. Tried everything, including another brand of cadence/speed sensor with the same hit or miss results. But once I shut the app down and bring up Wahoo Fitness, everything just works.

I am using a Sony Xperia XZ with Android 7.

**Comment by JHenneberg** [comment]
Hey,

I made the same experience. The sensors are found recently (1s ago) but they are not connected

Steps to reproduce:
1. Start record
2. `Settings -> sensors -> found` 
3. `Settings -> sensors -> used` 
4. `hear rate sensor; 1s ago, connecting ...`

I can't really tell when its happening.

**Comment by Sublimis** [comment]
Hi guys, did you have a chance to try the new beta version and see if this has been fixed?

**Comment by JHenneberg** [comment]
I saw it already in the changelog 👍 . I will go for a spinn early tomorrow morning and will test it.

**Comment by ccharn** [comment]
Did quite a bit testing this morning by spinning the bike indoors. Firstly, with 5.100 the battery optimisation warning message keeps popping up and I had to go into settings to disable battery optimisation even though it was already done in the previous session. This is a new issue. 

Coming back to the BLE sensor connection issue - Sensors are able to be found, selected, configured and put to use successfully. However, when I simulated a new session (phone restarted), not all the 3 sensors (hrm, speed, cadence) would connect. Those not connecting have to be removed, rediscovered, selected, and reconfigured in order to use them again.

Thank you.

**Comment by anton-vakulenko** [comment]
I confirm this issue. On new session only first two sensors will connect (cadence and speed). Third sensor (HR) have to be removed, rediscovered, selected, and reconfigured. Otherwise you see this: "hear rate sensor; 1s ago, connecting ..." and nothing more. I think this issue appeared after last (or the one before last) update (but I'm not sure). In Strava app the HR sensors connects in 3 seconds without issues...

Dear developers, it's terribly annoying and I thinking about switching to some other bike app (dispite I have paid version of UrbanBiker and really LOVE it). 

Redmi Note 8 pro
miui 12.0.8
android 10 qp1a.190711.020
bluetooth 5.0
